types/next-server/test/next-server-dynamic-tests.tsx

```diff
@@ -17,30 +17,42 @@ const LoadingComponent: React.StatelessComponent<LoadingComponentProps> = ({
 }) => <p>loading...</p>;
 
 // 1. Basic Usage (Also does SSR)
-const DynamicComponent = dynamic(asyncComponent);
+const DynamicComponent = dynamic(Promise.resolve(MyComponent));
 const dynamicComponentJSX = <DynamicComponent foo="bar" />;
 
-// 1.1 Basic Usage (Loader function)
-const DynamicComponent2 = dynamic(() => asyncComponent);
+// 1.1 Basic Usage (Loader function, module shape with 'export = Component' / 'module.exports = Component')
+const DynamicComponent2 = dynamic(() => Promise.resolve(MyComponent));
 const dynamicComponent2JSX = <DynamicComponent2 foo="bar" />;
 
+// 1.2 Basic Usage (Loader function, module shape with 'export default Component')
+const DynamicComponent3 = dynamic(() => Promise.resolve({ default: MyComponent }));
+const dynamicComponent3JSX = <DynamicComponent3 foo="bar" />;
+
+// TODO: Work with module shape 'export { Component }'
```

Comment:
how is this supposed to work? isn't the user supposed to resolve the component manually?

Reply:
`react-loadable` (of which `next/dynamic` is based off of since Next.js 7) supports the following import style:

https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/react-loadable/test/index.tsx#L40-L50

I haven't yet to test if it actually works for Next.js, but we couldn't actually do it with the way our types are structured. Besides, it might be better that we stick to default dynamic imports instead.

Reply:
Interesting, I guess we have to export `Component` from the module instead of `MyComponent` then?

Reply:
@joaovieira Yeah, that's the use for `./imports/no-default.tsx`.